### PR TITLE
f-cookie-banner@3.1.0 - Fix re-open link positioning

### PR DESCRIPTION
**Pull request by AndyMayJE**
### Fixed
- Reopen link positioning bug by adding and using a ResizeObserver. This ensures body content changes ouside of f-cookie-banner trigger and recalculates if the link should be absolutely positioned.

### Added
- New prop `shouldAbsolutePositionReopenLink` so teams or 3rd parties using this component can disable the re-open link absolute positioning

image

image

image

## Browsers Tested

- [x] Chrome (latest)
- [x] Safari iPhone 12 mini (not Checkout)
- [x] Edge 94 Windows 10
- [x] Chrome Samsung Galaxy S10

### CODE REVIEW

File: packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue [hunk 1]
```diff
@@ -190,9 +203,24 @@ export default {
...
+            this.bodyObserver.observe(bodyElement);
+        }
     },
 
     methods: {
+        /**
+         * Triggered by <body> ResizeObserver calculates if the body height less than the window
+         */
+        updateIsBodyHeightLessThanWindowHeight () {
+            const ropenElementHeight = this.$refs.reopenCookieBannerLink.$el.clientHeight || 0;
```
Comment (DamianMullins) [inline packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue]: Is the `r` at the start of the variable name supposed to be there?

File: packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue [hunk 2]
```diff
@@ -173,13 +182,17 @@ export default {
...
         }
+
     },
 
     watch: {
         isHidden (newVal) {
             this.shouldHideBanner = !!newVal;
         }
     },
+    destroyed () {
```
Comment (DamianMullins) [inline packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue:194]: NIT: whitespace

```suggestion

    destroyed () {
```

File: packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue [hunk 3]
```diff
@@ -190,9 +203,24 @@ export default {
             });
         }
         this.isIosBrowser = /(iPhone|iPad).*Safari/.test(navigator.userAgent);
+
+        if (this.shouldAbsolutePositionReopenLink) {
+            this.bodyObserver = new ResizeObserver(this.updateIsBodyHeightLessThanWindowHeight);
+            const bodyElement = document.documentElement || document.body;
```
Comment (DamianMullins) [inline packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue:210]: Does this pass SSR without the `typeof window === 'object'` style check?

File: packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue (hunk 1 above)
Comment (AndyMayJE) [reply packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue]: Ooops thanks

File: packages/components/organisms/f-cookie-banner/package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@justeat/f-cookie-banner",
   "description": "Fozzie Cookie Banner – Cookie Banner",
-  "version": "2.2.1",
+  "version": "2.2.2",
```
Comment (DamianMullins) [inline packages/components/organisms/f-cookie-banner/package.json]: I think we should bump the minor version as a new prop has been introduced as well as a fix?

Comment (AndyMayJE) [reply packages/components/organisms/f-cookie-banner/package.json]: Yes, I was thinking the same. Will update. Thanks

**Review by simonsandersje — DISMISSED** [review]

**Review by DamianMullins — DISMISSED** [review]

**Review by dandel10n — DISMISSED** [review]

File: packages/components/organisms/f-cookie-banner/CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,16 @@
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+v2.3.1
```
Comment (dandel10n) [inline packages/components/organisms/f-cookie-banner/CHANGELOG.md]: I haven't noticed it at first, but the version should become 3.1.0

**Review by DamianMullins — APPROVED** [review]

**Review by simonsandersje — APPROVED** [review]

**Pull request merged**